cmd/context/context.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2021 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package context
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/okteto/okteto/cmd/utils"
+	"github.com/okteto/okteto/pkg/analytics"
+	okContext "github.com/okteto/okteto/pkg/cmd/context"
+	k8Client "github.com/okteto/okteto/pkg/k8s/client"
+	"github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/okteto"
+	"github.com/spf13/cobra"
+)
+
+type ContextOptions struct {
+	Token       string
+	Namespace   string
+	clusterType okteto.ClusterType
+}
+
+// Context points okteto to a cluster.
+func Context() *cobra.Command {
+	ctxOptions := &ContextOptions{}
+	cmd := &cobra.Command{
+		Use:     "context [url|k8s-context]",
+		Aliases: []string{"ctx"},
+		Args:    utils.MaximumNArgsAccepted(1, "https://okteto.com/docs/reference/cli/#context"),
+		Short:   "Points okteto to a cluster",
+		Long: `Points okteto to a cluster
+
+Run
+    $ okteto context
+
+and this command this will ask you for the name of your kubernetes context.
+If an okteto cluster is selected it will open your browser to ask your authentication details and retrieve your API token. You can script it by using the --token parameter.
+
+By default, this will ask you for the kubernetes context you want to operate.
+
+If you want to log into your Okteto Enterprise instance, specify a URL. For example, run
+
+    $ okteto context https://okteto.example.com
```

Comment:
I may be saying the same thing as Ramiro but I am finding this command a little confusing. This may be due to the overloaded term `context` and that there are multiple types of parameters that I think won't work in cases I hit commonly. 

If my current `kube context` is `cluster-a` and I have a namespace `jacob`. What happens when I run `kube context jacob` and I wanted it to be `cluster-b` with namespace `jacob`? We can't guarantee uniqueness and I imagine I'll mess it up?

Reply:
In your example, do you mean `kube context`, or `okteto context`?
In other CLI tools in the space, like kubectl or docker, `context` sets your working cluster. Namespaces are a secondary attribute.

Reply:
I mean okteto context. As far as I can tell an okteto context = cluster+namespace and but it seems like the namespace is used to define a context and I'm wondering about possible collisions?